src/testdir/test_signs.vim

```diff
@@ -104,6 +104,27 @@ func Test_sign()
   exe 'sign jump 43 file=' . fn
   call assert_equal('B', getline('.'))
 
+  " You can't define a sign with a non-printable character as text
+  call assert_fails( "sign define Sign4 text= linehl=Comment", 'E239:' )
+  call assert_fails( "sign define Sign4 text=a linehl=Comment", 'E239:' )
+  call assert_fails( "sign define Sign4 text=a linehl=Comment", 'E239:' )
```

Comment:
It's not obvious that there is a special characters \x1b here. 
How about using \x1b in the string to be more explicit?

Reply:
Yes, that's a good shout.

Reply:
Done. I also added a test to show that `\\` is accepted.